scio-elasticsearch/src/main/java/com/spotify/scio/elasticsearch/ElasticsearchIO.java

```diff
@@ -0,0 +1,228 @@
+package com.spotify.scio.elasticsearch;
+
+import org.apache.beam.runners.dataflow.options.DataflowPipelineOptions;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.GroupByKey;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.windowing.AfterProcessingTime;
+import org.apache.beam.sdk.transforms.windowing.GlobalWindows;
+import org.apache.beam.sdk.transforms.windowing.Repeatedly;
+import org.apache.beam.sdk.transforms.windowing.Window;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PDone;
+import com.twitter.jsr166e.ThreadLocalRandom;
+import java.io.IOException;
+import java.io.Serializable;
+import java.net.InetSocketAddress;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+import org.elasticsearch.action.bulk.BulkRequestBuilder;
+import org.elasticsearch.action.bulk.BulkResponse;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.client.transport.TransportClient;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.transport.InetSocketTransportAddress;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ElasticsearchIO {
+  public static class Write {
+    /**
+     * Returns a tranform for writing to Elasticsearch cluster for a given name.
+     *
+     * @param clusterName name of the Elasticsearch cluster
+     */
+    public static Bound withClusterName(String clusterName) {
+      return new Bound().withClusterName(clusterName);
+    }
+
+    /**
+     * Returns a transform for writing to the Elasticsearch cluster for a given servers.
+     *
+     * @param servers endpoints for the Elasticsearch cluster
+     */
+    public static Bound withServers(InetSocketAddress[] servers) {
+      return new Bound().withServers(servers);
+    }
+
+    /**
+     * Returns a transform for writing to Elasticsearch cluster by providing
+     * slight delay specified by flushInterval.
+     *
+     * @param flushInterval delay applied to buffer elements. Defaulted to 1 seconds.
+     */
+    public static Bound withFlushInterval(Duration flushInterval) {
```

Comment:
Any reason why you are using `java.time` instead of `joda.time`. Both scio and beam use `joda`, this would also save you on converting java to joda below.